bottlecap/src/metrics/datadog.rs

```diff
@@ -37,38 +37,42 @@ impl DdApi {
         DdApi {
             api_key,
             site,
-            ureq_agent: ureq::AgentBuilder::new().build(),
+            client: reqwest::Client::new(),
         }
     }
 
     /// Ship a serialized series to the API, blocking
-    pub fn ship_series(&self, series: &Series) -> Result<(), ShipError> {
+    pub async fn ship_series(&self, series: &Series) -> Result<(), ShipError> {
         let body = serde_json::to_vec(&series)?;
         debug!("sending body: {:?}", &series);
 
         let url = format!("https://api.{}/api/v2/series", &self.site);
-        let resp: Result<ureq::Response, ureq::Error> = self
-            .ureq_agent
+        let resp = self
+            .client
             .post(&url)
-            .set("DD-API-KEY", &self.api_key)
-            .set("Content-Type", "application/json")
-            .send_bytes(body.as_slice());
+            .header("DD-API-KEY", &self.api_key)
+            .header("Content-Type", "application/json")
+            .body(body)
+            .send()
+            .await;
+
         match resp {
-            Ok(_resp) => Ok(()),
-            Err(ureq::Error::Status(code, response)) => Err(ShipError::Failure {
-                status: code,
-                body: response
-                    .into_string()
-                    .expect("body could not be converted to string"),
-            }),
+            Ok(resp) => match resp.status() {
+                reqwest::StatusCode::OK => Ok(()),
```

Comment:
```suggestion
```

Do we really need it, if the API will always return either 201/202?

Reply:
Removed